### PR TITLE
apache-airflow: 2.3.4 -> 2.4.1

### PR DESCRIPTION
**Pull request by risicle**
###### Description of changes

https://nvd.nist.gov/vuln/detail/CVE-2022-40604
https://nvd.nist.gov/vuln/detail/CVE-2022-40754

Ran the update script, needed to add new dependency `configupdater`.

I noticed that some of the top-level python dependencies don't appear to be needed (e.g. `cached-property`, `iso8601` and particularly `numpy` which might lighten the dependency load) but don't know enough about the package to know whether these are used in some optional/other way not declared by the requirements, so have left them all in place.

I'm a bit confused about the `checkPhase` - the package goes to the trouble of setting up `pytestCheckHook` and flagging tests to exclude, but then overrides the whole `checkPhase` so pytest never gets run :shrug:.

Very much hoping for review from maintainers as no tests afaik cover e.g. the frontend.

No backport needed as airflow is proper broken on stable.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
Having trouble on darwin - `checkPhase` segfaults for me:

```
running install tests
2.4.1
DB: sqlite:////private/tmp/nix-build-python3.10-apache-airflow-2.4.1.drv-0/tmp.6cvgJHsjvz/unittests.db
[2022-10-02 19:23:11,831] {migration.py:204} INFO - Context impl SQLiteImpl.
[2022-10-02 19:23:11,832] {migration.py:207} INFO - Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
/nix/store/jcgcp815vw82ggs16fgjgcxkyv6hwvhp-stdenv-darwin/setup: line 1413: 43262 Segmentation fault: 11  airflow db init
```

File: pkgs/development/python-modules/configupdater/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchPypi {
+    inherit version;
+    pname = "ConfigUpdater";
+    sha256 = "sha256-RvDHTXPvpyN3Z2S0PJc59oBSSV3T1zQxnB0OtYUR8Vs=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.cfg \
+      --replace '--cov' '#--cov'
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/configupdater/default.nix:20]:
```suggestion
      --replace "--cov configupdater --cov-report term-missing" ""
```

**Review by SuperSandro2000 — APPROVED** [review]
otherwise diff wise LGTM, didn't test

File: pkgs/development/python-modules/configupdater/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/development/python-modules/configupdater/default.nix:20]: Yes I guess in a future version they could append something after `--cov` that's useful

**Review by gbpdt — APPROVED** [review]
The change looks good to me.

Re: the checkPhase, I agree that does sound weird. I'm pretty sure I saw tests being run, will try to verify at some point but I don't think that should block this change. Perhaps it has something to do with the checkPhase vs installCheckPhase stuff for Python packages.

Were you able to run the airflow web UI? That's a good check to do if possible, although it's obviously a manual step.

**Comment by risicle** [comment]
I'm not sure how useful a check of just the web ui is without any of the other machinery working around it.

Still, as the package is currently broken/not building, this will be an improvement on that.

**Comment by risicle** [comment]
FWIW tests seem to work if the `checkPhase` is turned into a `preCheck` and the `export HOME=$(mktemp -d)` is moved to the end of the `postInstall` (because it needs to be set before the `pythonImportsCheck` phase)

**Comment by gbpdt** [comment]
> I'm not sure how useful a check of just the web ui is without any of the other machinery working around it.

You can invoke the built-in bash and python operator examples and check they run, which seems like a worthwhile end-to-end test.

**Pull request merged**